framework/src/base/MooseApp.C

```diff
@@ -797,6 +796,34 @@ MooseApp::setupOptions()
       _action_warehouse.setFinalTask("split_mesh");
     }
     _action_warehouse.build();
+
+    _file_base_set_by_user = false;
+    if (!_check_input)
```

Comment:
Where was this logic before? It doesn't seem like it should belong right here. In fact it changes the intention of the free `else` statement at the bottom. Now the only way that we would get into that else is if `_check_input == true` but based on the nested `if` condition already in that `else` clause that wasn't the original intention so something has changed. I realize that this routine is a pretty big mess, but logically speaking, the right place for this new logic would be inside of that last `else` condition, or that last `else` condition should be changed to handle only the `_check_input` condition. Something isn't right with the way it is now.

Reply:
If I do not protect the code with this, I got a strange failure in running test related with benchmark tests. I am not quite familiar with this function honestly. Do you want to add a commit to clean this routine up or we can leave this as it is? I think at least we cleaned up the file base properly.

Reply:
Can you elaborate a bit more here? Tell me how to replicate? I'd really like to understand so we can at least document what's going on here.